### PR TITLE
fix(ui-select): avoid dropdown flickering on search

### PR DESCRIPTION
**Pull request by marco-jantke**
I hope commit message and subject are descriptive enough :) This is a fix for #1298 with no other side effects then the flickering.

### CODE REVIEW

**Comment by wesleycho** [comment]
This LGTM

File: src/uiSelectDirective.js [hunk 1]
```diff
@@ -331,9 +338,6 @@ uis.directive('uiSelect',
               return;
             }
 
-            // Hide the dropdown so there is no flicker until $timeout is done executing.
-            dropdown[0].style.opacity = 0;
```
Comment (user378230) [inline src/uiSelectDirective.js]: Removing this introduces a problem when dropdown is positioned up

http://plnkr.co/edit/rM3rdaekwqG8wN77Hgb6?p=preview

Enter text in select2 "up" and delete with `backspace` quickly (I typed `ada` to get most results).

With this line removed the select gets shifted/overlayed with the list as a flicker. (Comment/uncomment line ~1261)

We need to detect if dropdown is positioned up and apply this along with your additions on line 298.

**Comment by marco-jantke** [comment]
Thanks a lot for feedback! I will have a look tomorrow.

**Comment by marco-jantke** [comment]
I had a look again in this issue. Sorry for the time delay guys..

I see the point of @user378230, when you reduce your search and thus extend the list again, the dropdown flickers for a short moment below the placeholder and then positions up. Though I can not figure out what is actually causing this or how to detect it..

I set up a plunkr with a couple of log messages where I tried to figure this out. Maybe you can give some useful input where to continue the investigation.

Please have a look in this Plunkr on lines 1259 and following. http://plnkr.co/edit/7NGlK9SRi95jNYTqpX1D?p=preview

**Comment by user378230** [comment]
@marco-jantke a shot in the dark but I wonder if the changes in #1132 help at all?

**Comment by rpocklin** [comment]
This fixes the flicker on search, but there was a reason this was added in the first place 
```// Hide the dropdown so there is no flicker until $timeout is done executing.```

The goal should be to avoid the opacity reset if the dropdown is already visible, i've created a tweaked PR in #1594

**Pull request closed without merging**